CURLINFO_FILETIME_T not available in ubuntu 18.04

`CURLINFO_FILETIME_T` was added to `curl` in v. `7.59.0`. 
Latest `curl` version available on Ubuntu 18.04 is `7.58.0`, hence `rtracklayer` install fails on Ubuntu 18.04. 

Some systems still rely on Ubuntu 18.04. 
FYI, `rtracklayer` install works on Ubuntu 18.04 by replacing the only occurence of `CURLINFO_FILETIME_T` by `CURLINFO_FILETIME` (here: https://github.com/lawremi/rtracklayer/blob/master/src/ucsc/net.c#L16). I don't know anything about `curl` so unsure how this would impact `rtracklayer`. 

Of note, install also works by manually installing `curl 7.59.0`, of course. 

This issue should be closed immediately if you are not planning on supporting Ubuntu 18.04 anymore. Otherwise let me know, I'd be happy to open a PR to change this. 

**`BiocManager::install("rtracklayer")` output:**
```
'getOption("repos")' replaces Bioconductor standard repositories, see 'help("repositories", package = "BiocManager")' for details.
Replacement repositories:
    CRAN: https://cloud.r-project.org
Bioconductor version 3.19 (BiocManager 1.30.23), R 4.4.0 (2024-04-24)
Installing package(s) 'rtracklayer'
trying URL 'https://bioconductor.org/packages/3.19/bioc/src/contrib/rtracklayer_1.64.0.tar.gz'
Content type 'application/x-gzip' length 3951281 bytes (3.8 MB)
==================================================
downloaded 3.8 MB

* installing *source* package 'rtracklayer' ...
** using staged installation
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for OPENSSL... yes
configure: creating ./config.status
config.status: creating src/Makevars
** libs
using C compiler: 'gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0'

[...]

gcc -I"/usr/share/R/include" -DNDEBUG -DUSE_SSL -D_FILE_OFFSET_BITS=64 -I'/home/rsg/R/x86_64-pc-linux-gnu-library/4.4/S4Vectors/include' -I'/home/rsg/R/x86_64-pc-linux-gnu-library/4.4/IRanges/include' -I'/home/rsg/R/x86_64-pc-linux-gnu-library/4.4/XVector/include'     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-U1b9am/r-base-4.4.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ucsc/errAbort.c -o ucsc/errAbort.o
ucsc/errAbort.c: In function 'getThreadVars':
ucsc/errAbort.c:255:5: warning: ignoring return value of 'write', declared with attribute warn_unused_result [-Wunused-result]
     write(STDERR_FILENO, errMsg, strlen(errMsg));
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gcc -I"/usr/share/R/include" -DNDEBUG -DUSE_SSL -D_FILE_OFFSET_BITS=64 -I'/home/rsg/R/x86_64-pc-linux-gnu-library/4.4/S4Vectors/include' -I'/home/rsg/R/x86_64-pc-linux-gnu-library/4.4/IRanges/include' -I'/home/rsg/R/x86_64-pc-linux-gnu-library/4.4/XVector/include'     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-U1b9am/r-base-4.4.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ucsc/hash.c -o ucsc/hash.o

[...]

gcc -I"/usr/share/R/include" -DNDEBUG -DUSE_SSL -D_FILE_OFFSET_BITS=64 -I'/home/rsg/R/x86_64-pc-linux-gnu-library/4.4/S4Vectors/include' -I'/home/rsg/R/x86_64-pc-linux-gnu-library/4.4/IRanges/include' -I'/home/rsg/R/x86_64-pc-linux-gnu-library/4.4/XVector/include'     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-U1b9am/r-base-4.4.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ucsc/udc.c -o ucsc/udc.o
gcc -I"/usr/share/R/include" -DNDEBUG -DUSE_SSL -D_FILE_OFFSET_BITS=64 -I'/home/rsg/R/x86_64-pc-linux-gnu-library/4.4/S4Vectors/include' -I'/home/rsg/R/x86_64-pc-linux-gnu-library/4.4/IRanges/include' -I'/home/rsg/R/x86_64-pc-linux-gnu-library/4.4/XVector/include'     -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-U1b9am/r-base-4.4.0=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ucsc/net.c -o ucsc/net.o
In file included from /usr/include/x86_64-linux-gnu/curl/curl.h:2739:0,
                 from ucsc/net.c:7:
ucsc/net.c: In function 'header_get_last_modified':
ucsc/net.c:16:47: error: 'CURLINFO_FILETIME_T' undeclared (first use in this function); did you mean 'CURLINFO_FILETIME'?
     CURLcode status = curl_easy_getinfo(curl, CURLINFO_FILETIME_T, &last_modified);
                                               ^
ucsc/net.c:16:47: note: each undeclared identifier is reported only once for each function it appears in
make: *** [/usr/lib/R/etc/Makeconf:195: ucsc/net.o] Error 1
ERROR: compilation failed for package 'rtracklayer'
* removing '/home/rsg/R/x86_64-pc-linux-gnu-library/4.4/rtracklayer'

The downloaded source packages are in
        '/tmp/Rtmpj7uAy8/downloaded_packages'
Warning message:
In install.packages(...) :
  installation of package 'rtracklayer' had non-zero exit status
```

**`BiocManager::valid()` output:**
```
'getOption("repos")' replaces Bioconductor standard repositories, see 'help("repositories", package = "BiocManager")' for details.
Replacement repositories:
    CRAN: https://cloud.r-project.org

* sessionInfo()

R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 18.04.6 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so;  LAPACK version 3.7.1

locale:
 [1] LC_CTYPE=en_IN.utf8        LC_NUMERIC=C               LC_TIME=en_IN.utf8         LC_COLLATE=en_IN.utf8      LC_MONETARY=en_IN.utf8    
 [6] LC_MESSAGES=en_IN.utf8     LC_PAPER=fr_FR.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=fr_FR.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/Paris
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] BiocManager_1.30.23 compiler_4.4.0      tools_4.4.0        

Bioconductor version '3.19'

  * 1 packages out-of-date
  * 0 packages too new

create a valid installation with

  BiocManager::install("microbiomeDataSets", update = TRUE, ask = FALSE, force = TRUE)

more details: BiocManager::valid()$too_new, BiocManager::valid()$out_of_date

Warning message:
1 packages out-of-date; 0 packages too new 
```

--- a/src/ucsc/net.c
+++ b/src/ucsc/net.c
@@ -12,21 +12,15 @@
 #include "cheapcgi.h"
 
 time_t header_get_last_modified(CURL *curl) {
-    CURLcode status;
     curl_off_t last_modified;
 
-    curl_version_info_data *ver = curl_version_info(CURLVERSION_NOW);
-    unsigned int major = (ver->version_num >> 16) & 0xff;
-    unsigned int minor = (ver->version_num >> 8) & 0xff;
-    unsigned int patch = ver->version_num & 0xff;
+    #if LIBCURL_VERSION_NUM >= 0x073b00
+        #define FILETIME CURLINFO_FILETIME_T
+    #else
+        #define FILETIME CURLINFO_FILETIME
+    #endif
 
-    if (major > 7)
-        status = curl_easy_getinfo(curl, CURLINFO_FILETIME_T, &last_modified);
-    if (major == 7 && minor >= 59 && patch >= 0)
-        status = curl_easy_getinfo(curl, CURLINFO_FILETIME_T, &last_modified);
-    else
-        status = curl_easy_getinfo(curl, CURLINFO_FILETIME, &last_modified);
-
+    CURLcode status = curl_easy_getinfo(curl, FILETIME, &last_modified);
 
     if ((CURLE_OK == status) && (last_modified >= 0)) {
         struct tm *utc_tm_info = gmtime(&last_modified);

--- a/src/ucsc/net.c
+++ b/src/ucsc/net.c
@@ -12,8 +12,21 @@
 #include "cheapcgi.h"
 
 time_t header_get_last_modified(CURL *curl) {
+    CURLcode status;
     curl_off_t last_modified;
-    CURLcode status = curl_easy_getinfo(curl, CURLINFO_FILETIME_T, &last_modified);
+
+    curl_version_info_data *ver = curl_version_info(CURLVERSION_NOW);
+    unsigned int major = (ver->version_num >> 16) & 0xff;
+    unsigned int minor = (ver->version_num >> 8) & 0xff;
+    unsigned int patch = ver->version_num & 0xff;
+
+    if (major > 7)
+        status = curl_easy_getinfo(curl, CURLINFO_FILETIME_T, &last_modified);
+    if (major == 7 && minor >= 59 && patch >= 0)
+        status = curl_easy_getinfo(curl, CURLINFO_FILETIME_T, &last_modified);
+    else
+        status = curl_easy_getinfo(curl, CURLINFO_FILETIME, &last_modified);
+
 
     if ((CURLE_OK == status) && (last_modified >= 0)) {
         struct tm *utc_tm_info = gmtime(&last_modified);

